### PR TITLE
replace push_back with emplate_back

### PR DESCRIPTION
**Pull request by neheb**
The latter forwards the arguments directly.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: src/builtin_argparse.cpp [hunk 1]
```diff
@@ -535,9 +535,9 @@ static int handle_flag(parser_t &parser, const argparse_cmd_opts_t &opts, option
         // short or long flag was given.
         assert(!woptarg);
         if (long_idx == -1) {
-            opt_spec->vals.push_back(wcstring(1, L'-') + opt_spec->short_flag);
+            opt_spec->vals.emplace_back(1, L'-' + opt_spec->short_flag);
```
Comment (faho) [inline src/builtin_argparse.cpp:538]: I don't think this is correct - the original was just a cheesy way of saying "Make a string containing `-`", via the "fill" constructor. I'm not sure which constructor this is now supposed to use, but it appears to not be working - see the failing tests.

**Review by beryllium76 — APPROVED** [review]

**Review by faho — CHANGES_REQUESTED** [review]
It borken.

File: src/builtin_argparse.cpp (hunk 1 above)
Comment (neheb) [reply src/builtin_argparse.cpp:538]: Right. It's not working because of thr 1. I'll fix.

**Comment by neheb** [comment]
Fixed. Added some extra.

**Comment by mqudsi** [comment]
lgtm, all seem to be valid uses of `emplace_back()`. That said, the benefit probably wouldn't show in even a micro benchmark after all automated optimizations are said and done.

**Comment by neheb** [comment]
Probably not. emplace_back is just shorter.

**Comment by mqudsi** [comment]
Thanks, merged.

**Pull request merged**